src/aspire/noise/noise.py

```diff
@@ -345,3 +354,297 @@ def estimate_noise_psd(self):
+        """
+
+        if bg_radius is None:
+            bg_radius = (np.floor(src.L / 2) - 1) / src.L
+
+        super().__init__(src=src, bg_radius=bg_radius, batch_size=batch_size)
+
+        self.max_d = max_d
+        if self.max_d is None:
+            self.max_d = np.floor(src.L / 3) / src.L
+
+    def estimate(self):
+        """
+        :return: The estimated noise variance of the images.
+        """
+
+        return np.mean(self.filter.evaluate_grid(self.src.L))
+
+    def _create_filter(self):
+        """
+        :return: The estimated noise power spectral distribution (PSD) of the images in the form of a filter object.
+        """
+        return ArrayFilter(self._estimate_noise_psd())
+
+    def _estimate_noise_psd(self):
+        """
+        :return: The estimated noise variance of the images in the Source used to create this estimator.
+        """
+        # Setup parameters
+        samples_idx = grid_2d(self.src.L, normalized=False, shifted=True)["r"] >= (
+            self.bg_radius * self.src.L
+        )
+        max_d_pixels = round(self.max_d * self.src.L)
+
+        psd = self.estimate_power_spectrum_distribution_2d(
+            self.src.images,
+            samples_idx,
+            max_d_pixels,
+            batch_size=self.batch_size,
+        )[0]
+
+        return psd
+
+    @staticmethod
+    def estimate_power_spectrum_distribution_1d(
+        images, samples_idx, max_d=None, batch_size=512
+    ):
+        """
+        Estimate the 1D isotropic autocorrelation function of `images`.
+        The samples to use in each image are given by `samples_idx` mask.
+        The correlation is computed up to a maximal distance of `max_d`.
+
+        Port of MATLAB `cryo_epsdR`.
+
+        :param images: `Image` instance
+        :param samples_idx: Boolean mask shaped `(L,L)`.
+        :param max_d: Max computed correlation distance in pixels.
+           Default of `None` yields `np.floor(L / 3)`.
+        :param batch_size:  The size of the batches in which to compute the variance estimate.
+        :return:
+            - Radial PSD array of shape
+            - Distances map array
+            - Count of nonzero correlations array
+        """
+
+        n_img = images.n_images
+        L = samples_idx.shape[-1]
+        batch_size = min(batch_size, n_img)
+
+        # Correlations more than `max_d` pixels apart are not computed.
+        if max_d is None:
+            max_d = np.floor(L / 3)
+        if max_d > L - 1:
+            logger.info(
+                f"`max_d` value {max_d}greater than number of image pixels {L}, clipping to {L-1}."
+            )
+        max_d = int(min(max_d, L - 1))
+
+        # Compute distances
+        # Note grid_2d['r'] is not used because we want an integer grid directly;
+        #   yields integer dists (radius**2) values.
+        X, Y = xp.mgrid[0 : max_d + 1, 0 : max_d + 1]
+        dists = X * X + Y * Y
+        dsquare = xp.sort(xp.unique(dists[dists <= max_d**2]))
```

Comment:
`unique_dists` or `unique_dsquare` (with renaming `dists` to `dsquare`)?

Reply:
Sure. I agree with you, not a fan of the names, but my intention was to make comparing the codes as easy as possible....

Reply:
and without making then too slow in python I guess.

Reply:
Changed here and in the other method.